src/views/EventsList.vue

```diff
@@ -3,30 +3,48 @@
     <h1>This is Events List page</h1>
     <nav>    
       <ul>            
-        <router-link v-for="item of DataList.events" :key="item.id" 
-            :to="{name: 'event-item', 
-            params: {id: String(item.id), item: item}}">
-          Event Item #{{item.id}}
-        </router-link>         
-      </ul>           
-      <router-view/>
-      
+        <li v-for="item of DataList" :key="item.id">
+            <router-link  
+              :to="{name: 'event-item', 
+              params: {id: String(item.id), item: item}}">
+              Event Item #{{item.id}}          
+            </router-link>         
+            <button @click="removeItem(item)"> X </button>
+        </li>
+      </ul>                       
+      <router-view/>            
     </nav>
   </div>
 </template>
 
 <script>
-    import DataList from '../../db.json'
+    //import DataList from '../../db.json'
+    import store from '@/vuex/store'
```

Comment:
Do you really need to import store? You can point to $store object or use mapState helper